indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/X509SignatureAlgorithm.kt

```diff
@@ -91,35 +93,83 @@ enum class X509SignatureAlgorithm(
             ES256, PS256, RS256 -> Digest.SHA256
             ES384, PS384, RS384 -> Digest.SHA384
             ES512, PS512, RS512 -> Digest.SHA512
+            else -> throw IllegalArgumentException("Unsupported hash algorithm.")
         }
 
+    /**
+     * **Accessing this property may throw.**
+     * @throws IllegalArgumentException in case a non-supported [X509SignatureAlgorithm] is to be converted
+     */
+    // TODO update when core signature data classes become extensible
     override val algorithm: SignatureAlgorithm
```

Comment:
@iaik-jheher I really think we need to make the base property nullable to support arbitrary mappings. Same goes for SpecializedEncryptionAlgorithm

Reply:
@iaik-jheher  Do we also make `SpecializedSymmetricEncryptionAlgorithm`'s `algorithm` nullable while we're at it?